Documentation/botr/ryujit-overview.md

```diff
@@ -246,139 +248,132 @@ As the JIT has evolved, changes have been made to improve the ability to reason
+ t4 =    LCL_VAR   double V01 arg1         u:1 $81
+     ┌──▌  t3     double 
+     ├──▌  t4     double 
+ t5 = ▌  MUL       double $140
+ t7 =    LCL_VAR   double V00 arg0         u:1 $80
+ t6 =    CNS_DBL   double 4.0000000000000000 $180
+     ┌──▌  t7     double 
+     ├──▌  t6     double 
+ t8 = ▌  MUL       double $141
+ t9 =    LCL_VAR   double V02 arg2         u:1 $82
+     ┌──▌  t8     double 
+     ├──▌  t9     double 
+t10 = ▌  MUL       double $142
+     ┌──▌  t5     double 
+     ├──▌  t10    double 
+t11 = ▌  SUB       double $143
+     ┌──▌  t11    double 
+t12 = ▌  INTRINSIC double sqrt $83
+     ┌──▌  t12    double 
+      ▌  STORE_LCL_VAR double V06 cse0         
+t46 =    LCL_VAR   double V06 cse0          $83
+ t1 =    LCL_VAR   double V01 arg1         u:1 $81
+     ┌──▌  t1     double 
+ t2 = ▌  NEG       double $84
+     ┌──▌  t2     double 
+      ▌  STORE_LCL_VAR double V08 cse2         
+t56 =    LCL_VAR   double V08 cse2          $84
+     ┌──▌  t46    double 
+     ├──▌  t56    double 
+t13 = ▌  ADD       double $144
+t15 =    LCL_VAR   double V00 arg0         u:1 $80
+t14 =    CNS_DBL   double 2.0000000000000000 $181
+     ┌──▌  t15    double 
+     ├──▌  t14    double 
+t16 = ▌  MUL       double $145
+     ┌──▌  t16    double 
+      ▌  STORE_LCL_VAR double V07 cse1         
+t51 =    LCL_VAR   double V07 cse1          $145
+     ┌──▌  t13    double 
+     ├──▌  t51    double 
+t17 = ▌  DIV       double $146
+ t0 =    LCL_VAR   byref  V03 arg3         u:1 (last use) $c0
+     ┌──▌  t0     byref  
+     ├──▌  t17    double 
+      ▌  STOREIND  double
+```
 
 ## <a name="lowering"/>Lowering
 
 Lowering is responsible for transforming the IR in such a way that the control flow, and any register requirements, are fully exposed.
 
-It accomplishes this in two passes.
-
-The first pass is an execution-order traversal that performs context-dependent transformations such as expanding switch statements (using a switch table or a series of conditional branches), constructing addressing modes, etc.  For example, this:
-
-    t0   = lclVar     ref    V00 arg0
-    t1   = lclVar     int    V03 loc1
-           ┌──▌  t1   int
-    t2   = cast       long <- int
-    t3   = const      long   2
-           ┌──▌  t2   long
-           ├──▌  t3   long
-    t4   = <<         long
-           ┌──▌  t0   ref
-           ├──▌  t4   long
-    t5   = +          byref
-    t6   = const      long   16
-           ┌──▌  t5   ref
-           ├──▌  t6   long
-    t7   = +          byref
-           ┌──▌  t7   ref
-    t8   = indir      int
+It does an execution-order traversal that performs context-dependent transformations such as expanding switch statements (using a switch table or a series of conditional branches), constructing addressing modes, determining the code generation strategy for block assignments (e.g. `GT_STORE_BLK`) which may become helper calls, unrolled loops, or an instruction like `rep stos` etc.  For example, this:
```

Comment:
This should also mention the containedness analysis, which used to be part of the register allocator, but is now done as part of Lowering.

Reply:
Expanded to mention reg optional, containment and generating target specific instructions such as `BT`.